apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReadOrQueryRequest.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+
+namespace Google.Cloud.Spanner.V1
+{
+    /// <summary>
+    /// Interface for all common properties of <see cref="ReadRequest"/> and <see cref="ExecuteSqlRequest"/>.
+    /// </summary>
+    public interface IReadOrQueryRequest
```

Comment:
I don't like these being public, we'd be in trouble if the Read and Query requests (and all the other) start differing in the future. When I proposed this, I didn't fully realized we'd need them in both layers. Maybe

- Use another abstraction, like a private class that alternatively wraps either of the types, and we base the implementation on that class, and only return/receive the sql or read specific types on the public surface. Use the same approach in both layers, with one of these classes in each layer, containing what that layer needs only (we might need to duplicate some code there).
- Use the (private) interface approach in V1 and the class approach from the point above in Data.
- If we are going to keep the interfaces, let's at least move them to the Internal folder/namespace, so that we can give a hint to users that these are not to be used by them?

@jskeet WDYT?